src/main/webapp/app/admin/statistics/statistics-graph.component.ts

```diff
@@ -0,0 +1,215 @@
+import { Component, Input, OnChanges, OnInit, SimpleChanges, ViewChild } from '@angular/core';
+import { StatisticsService } from 'app/admin/statistics/statistics.service';
+import { ChartDataSets, ChartOptions, ChartType } from 'chart.js';
+import { BaseChartDirective, Label } from 'ng2-charts';
+import { DataSet } from 'app/exercises/quiz/manage/statistics/quiz-statistic/quiz-statistic.component';
+import { TranslateService } from '@ngx-translate/core';
+import * as moment from 'moment';
+import { Graphs, SpanType } from 'app/entities/statistics.model';
+
+@Component({
+    selector: 'jhi-statistics-graph',
+    templateUrl: './statistics-graph.component.html',
+})
+export class StatisticsGraphComponent implements OnInit, OnChanges {
+    // Inputs
+    @Input()
+    graphType: Graphs;
+    @Input()
+    currentSpan: SpanType;
+
+    // html properties
+    LEFT = false;
+    RIGHT = true;
+    SpanType = SpanType;
+    Graphs = Graphs;
+
+    // Histogram related properties
+    public barChartOptions: ChartOptions = {};
```

Comment:
Is it intentional that lines 28-38 contain the `public` accessor?

Reply:
I could argue with consistency, all our chart related variables are declared public. The compiler complains when stating the variable as private since it is used in the html template.I declare them package private now in [this](https://github.com/ls1intum/Artemis/pull/2511/commits/c16f4b7afac2d4b518fcd75ff6d28830a57ce8cf) commit